services/s3/src/main/java/software/amazon/awssdk/services/s3/parsing/S3Uri.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.parsing;
```

Comment:
It's not exactly just related to parsing. `model` subpackage is also not good because people may confuse it with service model POJOs. Maybe just under `software.amazon.awssdk.services.s3`? (sorry, just noticed it now)